### DISCUSSION TITLE
Join or merge two caches into one

### DISCUSSION THREAD

**Discussion by tenzap** [ccache/ccache#1413] [Questions & answers] [answered]
How would I join 2 ccache dirs into one?

Use case is having 2 CI build jobs running in parallel each producing ccache dir.

When both are finished, a 3rd job would run using a ccache dir containing the ccache of both previous jobs.

This question is for both ccache version 3 and 4 because the jobs run on systems with version 3.

**Comment by tenzap** [comment]
Seems one can copy each ccache dir into a 3rd ccache dir.
After that, to have correct counters, one would have to run `ccache -c`

Is that correct and supposed to work, or are there any problems that can come up?

**Comment by jrosdahl** [comment, marked as answer]
There is no official way to do that. However, as you already noted, you can just copy the files from one cache into the other and then run `ccache -c` to update the file number and size counters. There is no guarantee that it will keep working in the future, though. Also, you will lose some of the statistics counters (hits, misses, uncacheable calls, etc).

The above applies to both version 3.* and 4.*.

**Reply by tenzap** [reply]
Thanks for confirming. In case this won't work anymore in future, it would be nice if ccache was able to use multiple local ccache dirs simultaneously.

**Comment by stefanct** [comment]
Simply copying the files over is known to work (and I just did use that too) as described in more detail on the samba ML a few years ago: https://www.mail-archive.com/ccache@lists.samba.org/msg01416.html

In your case I would suggest configuring the CI jobs to work on the same dir from the beginning.
